text/0049-soc-gpio-peripheral.md

```diff
@@ -140,8 +154,9 @@ The members of a `gpio.PinSignature` are defined as follows:
 
 ```python3
 {
-    "i":  In(unsigned(1)),
-    "o":  Out(unsigned(1)),
+    "alt_mode": Out(unsigned(1)),
```

Comment:
This will make it impossible to use `connect()` with `gpio.PinSignature`, which breaks a common use case. (This is why I suggested `alt_mode` be a separate output on the side.)

Reply:
Ah right, I'll change it to `"alt_mode": Out(unsigned(pin_count))` on the `gpio.Peripheral` signature.